### PR TITLE
AKS: add support for ingress-appgw addon

### LINKED ISSUE #7384: Support for ingress-appgw add-on in azurerm_kubernetes_cluster
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AKS ability to deploy managed Ingress controller for Azure Application Gateway is now in preview as add-on. [https://docs.microsoft.com/en-us/azure/application-gateway/ingress-controller-overview](https://docs.microsoft.com/en-us/azure/application-gateway/ingress-controller-overview)

Would you consider adding to azurerm_kubernetes_cluster, please?

### New or Affected Resource(s)

* azurerm_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by derkoe**
Closes #7384

### CODE REVIEW

**Comment by Buthrakaur** [comment]
Is there any ETA on this to be merged and released, please?

**Comment by picardsrcd** [comment]
@tombuildsstuff do you have an ETA to having this merged in and released? 
From a workaround perspective, ill look at enabling the addon using CLI in the interim.

**Comment by mazen423** [comment]
That pull request needs attention to be deployed

File: azurerm/internal/services/containers/kubernetes_addons.go [hunk 1]
```diff
@@ -84,6 +87,39 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
 					},
 				},
 
+				"ingress_appgw": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_addons.go:90]: could we not abbreviate this?
```suggestion
				"ingress_app_gateway": {
```

File: azurerm/internal/services/containers/kubernetes_addons.go [hunk 2]
```diff
@@ -84,6 +87,39 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
...
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"enabled": {
+								Type:     schema.TypeBool,
+								Required: true,
+							},
+							"appgw_id": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_addons.go:100]:
```suggestion
							"gateway_id": {
```

File: azurerm/internal/services/containers/kubernetes_addons.go [hunk 3]
```diff
@@ -84,6 +87,39 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
...
+								Type:         schema.TypeString,
+								Optional:     true,
+								ValidateFunc: azureHelpers.ValidateResourceID,
+							},
+							"appgw_name": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_addons.go:105]: is it possible to pull this out of the gateway id?

File: azurerm/internal/services/containers/kubernetes_addons.go [hunk 4]
```diff
@@ -84,6 +87,39 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
...
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"appgw_subnet_id": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_addons.go:109]: given the block name containers app gateway could we
```suggestion
							"subnet_id": {
```

also is this something we could load from the gateway? or does it potentially have multiple?

File: azurerm/internal/services/containers/kubernetes_addons.go [hunk 5]
```diff
@@ -84,6 +87,39 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
...
+								Type:         schema.TypeString,
+								Optional:     true,
+								ValidateFunc: azureHelpers.ValidateResourceID,
+							},
+							"appgw_subnet_prefix": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_addons.go:114]:
```suggestion
							"subnet_prefix": {
```
also is this something we could load from the gateway? or does it potentially have multiple?

**Review by katbyte — CHANGES_REQUESTED** [review]
Hey @derkoe - thanks for the pr, however it appears to be missing a test for the new block and updates to the docs. Could we get those added?

**Comment by cwiederspan** [comment]
@derkoe Any chance this is still being worked on? I also have the same need and it looks like you're off to a great start to getting this going. I haven't been a contributor on this project yet, but I'm willing to try if there's anything I can do to help.

**Comment by derkoe** [comment]
@cwiederspan I'm sorry but we are not using Application Gateway anymore so my focus is not on this PR anymore. Maybe you can take over?

**Comment by magodo** [comment]
This feature has been picked up by #11376, for which I've just done a review. Probably people here can also subscribe that PR, which contains more comprehensive test and document.

**Comment by katbyte** [comment]
I'm going to close this in favour of #11376 as @derkoe is no longer working on this

**Pull request closed without merging**